src/autocomplete/RoomProvider.tsx

```diff
@@ -115,7 +124,7 @@ export default class RoomProvider extends AutocompleteProvider {
                     ),
                     range,
                 };
-            }).filter((completion) => !!completion.completion && completion.completion.length > 0).slice(0, 4);
```

Comment:
Hmm, why don't we need to slice any more?

Reply:
We honour a limit based on the argument provided - https://github.com/matrix-org/matrix-react-sdk/pull/6051/files#diff-f7d76bd7ba734cbae58aba1e1a7405611c4e739977a1b301b79b023357c21e71R107

This slice looks to have been a rooms-provider specific fix for performance issues, now we have a general case one we should be consistent with it